MapboxGeocoder/MBGeocoder.swift

```diff
@@ -269,42 +271,42 @@ public class Geocoder: NSObject {
         }
         
         let queryComponent = options.queries.map {
-            $0.stringByReplacingOccurrencesOfString(" ", withString: "+")
-                .stringByAddingPercentEncodingWithAllowedCharacters(NSCharacterSet.geocodingQueryAllowedCharacterSet()) ?? ""
-        }.joinWithSeparator(";")
+            $0.replacingOccurrences(of: " ", with: "+")
+                .addingPercentEncoding(withAllowedCharacters: CharacterSet.geocodingQueryAllowedCharacterSet()) ?? ""
+        }.joined(separator: ";")
         
-        let unparameterizedURL = NSURL(string: "/geocoding/v5/\(mode)/\(queryComponent).json", relativeToURL: apiEndpoint)!
-        let components = NSURLComponents(URL: unparameterizedURL, resolvingAgainstBaseURL: true)!
+        let unparameterizedURL = URL(string: "/geocoding/v5/\(mode)/\(queryComponent).json", relativeTo: apiEndpoint)!
+        var components = URLComponents(url: unparameterizedURL, resolvingAgainstBaseURL: true)!
```

Comment:
Yes: in Swift 3, URLComponents is a Swift struct, so `var` allows us to set `queryItems` below.